pkg/dns/serviceresolver.go

```diff
@@ -188,7 +193,7 @@ func (b *ServiceResolver) Records(dnsName string, exact bool) ([]msg.Service, er
 		// return endpoints
 		endpoints, err := b.endpoints.Endpoints(namespace).Get(name)
 		if err != nil {
-			return nil, err
+			return nil, errNoSuchName
```

Comment:
same question here, should errors other than an IsNotFound error trigger an errNoSuchName?